plugins/modules/entrust_cagw_certificate.py

```diff
@@ -0,0 +1,860 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c), Entrust Datacard Corporation, 2019
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: entrust_cagw_certificate
+author:
+    - Sapna Jain (@jains)
+short_description: Request SSL/TLS certificates with the Certificate Authority Gateway (CAGW) API
+description:
+    - Create, get, and take actions (Hold, Unhold, Revoke certificates) with the Certificate Authority Gateway (CAGW) API.
+    - Requires credentials for calling the CAGW API.
+notes:
+    - C(path) must be specified as the output location of the certificate.
+requirements:
+    - cryptography >= 1.6
+options:
+    force:
+        description:
+            - If force is used, a certificate is requested regardless of whether I(path) points to an existing valid certificate.
+        type: bool
+        default: False
+    path:
+        description:
+            - The destination path for the generated certificate as a PEM encoded cert.
+            - If there is already an Entrust certificate at this location, it will be replaced always.
+            - If enrollment_format is PKCS12 then it will have Base64 encoded PKCS12 body
+        type: path
+    csr:
+        description:
+            - Base-64 encoded Certificate Signing Request (CSR). csr is accepted without PEM formatting around the Base-64 string.
+            - If no csr is provided when request_type=new and enrollment_format=X509, the certificate will not be generated and module will be failed
+        type: path
+
+    cagw_api_client_cert_path:
+        description:
+            - Path for the Client cert issued by the same CA
+        type: path
+        required: true
+
+    cagw_api_client_cert_key_path:
+        description:
+            - Path for the Client cert key issued by the same CA
+        type: path
+        required: true
+
+    host:
+        description:
+            - Host or ip address for Entrust CAGW
```

Comment:
```suggestion
            - Host or IP address for Entrust CAGW.
```
Acronyms should always be upper-case (https://docs.ansible.com/ansible/devel/dev_guide/style_guide/spelling_word_choice.html#acronyms).

Reply:
@felixfontein Thanks a lot for your comments. 
Yeah I need to fix the failed sanity tests and other documentation related issues which I will take up as soon as possible.